tutorial/part-1-creating-a-simple-view/creating-your-first-table.md

```diff
@@ -0,0 +1,141 @@
+{% endhint %}
+
+## Accessing query results from a widget
+
+The next step is to display the query results in the **Products\_Table** :
+
+1. Navigate to **ProductListPage → Widgets → Products\_Table**
+2. Open **Product\_Table’**s properties
+3. Type the following in the Table Data property: `{{ProductsQuery.data}}`
+4. Your table now displays ProductsQuery’s results
+5. Click the Deploy button on the top right
+6. Click on the down arrow button next to the deploy button to get the URL of your app
+
+Visit the app URL with your web browser. You'll see a page displaying your query results in a table like below:  
+
+
+![](https://lh5.googleusercontent.com/wjbhU2Nsq_tfEFoAsI4qEn60jo6E8dkySMMUqoV9h1IdfBJ9Ug48_EkI-LZVaRK3VB4ebTi0OTbYFczticODH13A-XWJi-qhE12Lhz8OSXnCvRDB6uqceArq3wDVZA5xOaQlAogK)
+
+Let’s see how this works:
+
+1. By using mustaches, you’re asking Appsmith to resolve anything within it as JavaScript code.
+2. By using the query’s name within mustaches `{{ ProductsQuery }}`, you’re accessing the entire query object of **ProductsQuery**.
+3. By calling the data function on **ProductsQuery**, you’re accessing its results. 
+4. By setting the **Table Data** to `{{ ProductsQuery.data }}`, you’re telling **Products\_Table** to display the results of **ProductsQuery**.
+
+Note that you are accessing a query's result in a widget by calling a method on the query's name. As you'll see in the next section, the inverse of this is also possible, i.e a widget's state can also be accessed by a query. Furthermore, all the building blocks of an Appsmith page - Widgets, DB Queries, and APIs can access each other's data and/or state using their names. 
+
+{% hint style="info" %}
+Think of widgets, APIs and DB Queries in Appsmith as variables in your programming language. Similar to variables:
+
+* They are identified by a name
+* They represent an object, be it a widget, an API object, or a query object
+* They support a set of methods
+* They have a scope; they can be accessed from only within their parent page
+{% endhint %}
+
+{% hint style="info" %}
+**Names & Scope:**
+
+All names within a page must be unique - be it widget names, query names, or API names. 
+{% endhint %}
+
+{% hint style="info" %}
+**Application Hosting your application:**
+
+By default, the deployed app is hosted on Appsmith's domain, i.e. app.appsmith.com. To host your app on your custom domain,  read the detailed steps [here](https://docs.appsmith.com/quick-start), depending on where you've set up Appsmith.
+{% endhint %}
+
+## What's next?
+
+When you’re comfortable with the basics of using a display-widget, setting up a DB query, and connecting the widget to display query results, read [part 2 ](https://app.gitbook.com/@appsmith/s/appsmith/~/drafts/-MNXsPmxVacsRbqB7S_f/v/v1.3/tutorial/part-2-creating-a-basic-form)of the tutorial to learn to use widgets that enable you to take and process user input.
```

Comment:
One main concern I have is will a user who reads this be able to connect a query to a dropdown? This is one of the problems we have where users are not able to bridge that gap because the dropdown requires an array of <label, value>. Will we cover this later somehow?
I feel for only viewing some information around how to filter / search / paginate this table is missing. We haven't explained the reactive paradigm. i.e dropdown / searchbox calls query and automatically the table refreshes. This mental model users really struggle with

Reply:
> One main concern I have is will a user who reads this be able to connect a query to a dropdown? This is one of the problems we have where users are not able to bridge that gap because the dropdown requires an array of <label, value>. Will we cover this later somehow?
The two concerns here are about -.1.extending knowledge of what how you displayed data in table to showing options in a dropdown, and much more 2. Data transformation
I've addressed the first more explicitly now in part 1 of the tutorial itself. The second one is addressed in the data transformation section in part 2 of the tutorial.

Reply:
> We haven't explained the reactive paradigm. i.e dropdown / searchbox calls query and automatically the table refreshes. This mental model most users really struggle with

I've added this in part 1 where table shows query data, and also referenced it again in part 3 when table data needs to be refreshed after e edit product is successful.